### PR TITLE
Update for external load balancer.

### LINKED ISSUE #4411: Bringing down the cluster doesn't delete forwarding rules and load balancer for default services
Bringing down the cluster doesn't delete forwarding rules and load balancer for default services running on GCE.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Implementation of updating external load balancer on service update. This partially fixes #4411.

### CODE REVIEW

File: pkg/registry/service/rest.go [hunk 1]
```diff
@@ -276,3 +300,21 @@ func (rs *REST) deleteExternalLoadBalancer(service *api.Service) error {
 	}
 	return nil
 }
+
+func externalLoadBalancerNeedsUpdate(old, new *api.Service) bool {
+	if old.Spec.CreateExternalLoadBalancer != new.Spec.CreateExternalLoadBalancer ||
```
Comment (thockin) [inline pkg/registry/service/rest.go]: Should this also check (old.Spec.CreateExternalLoadBalancer || new.Spec.CreateExternalLoadBalancer) ?  If neither spec has a load balancer, you can safely say "false" right?

File: pkg/registry/service/rest.go [hunk 2]
```diff
@@ -228,7 +191,21 @@ func (rs *REST) Update(ctx api.Context, obj runtime.Object) (runtime.Object, boo
 	if errs := validation.ValidateServiceUpdate(oldService, service); len(errs) > 0 {
 		return nil, false, errors.NewInvalid("service", service.Name, errs)
 	}
-	// TODO: check to see if external load balancer status changed
+	// Recreate external load balancer if changed.
+	if externalLoadBalancerNeedsUpdate(oldService, service) {
```
Comment (thockin) [inline pkg/registry/service/rest.go]: TODO: support updating existing balancers

**Comment by thockin** [comment]
Overall this looks fine.  I'm going to throw this at Quinton because a) he should get more familiar with it and b) I am short on time this week, so I would like a second set of eyes. :)

@quinton-hoole-google

File: pkg/registry/service/rest.go (hunk 1 above)
Comment (jszczepkowski) [inline pkg/registry/service/rest.go]: Yes, added.

File: pkg/registry/service/rest.go (hunk 2 above)
Comment (jszczepkowski) [inline pkg/registry/service/rest.go]: Done.

**Comment by jszczepkowski** [comment]
All comments applied, PTAL.

**Comment by satnam6502** [comment]
**/sub** (because I am also dealing with this code as part of the P1 Fixit for an end to end test for cluster logging)

**Comment by quinton-hoole-google** [comment]
I've been through this code in a reasonable amount of detail and can't see any problems either.  LGTM.

**Pull request merged**